sympy/matrices/common.py

```diff
@@ -203,18 +203,24 @@ def col_insert(self, pos, other):
         if not self:
             return type(self)(other)
 
-        if pos < 0:
-            pos = self.cols + pos
-        if pos < 0:
-            pos = 0
-        elif pos > self.cols:
-            pos = self.cols
+        try:
+            as_int(pos)
+        except ValueError:
+            raise IndexError("Index is not an integer: 'pos = %s', valid : %s must be an integer" % (
+            pos, pos))
```

Comment:
`IndexError` is for out-of-bound errors, `TypeError` should be used here (that too if an exception has to be raised explicitly).